src/Daemon/Daemon.cpp

```diff
@@ -460,6 +466,51 @@ namespace usbguard
   {
   }
 
+  pid_t Daemon::pid_fd = -1;
+
+  void Daemon::daemonize(const std::string &pid_file)
+  {
+      USBGUARD_LOG(Trace) << "Starting daemonization";
+
+      pid_t pid = 0;
+
+      USBGUARD_SYSCALL_THROW("Daemonize", (pid = fork()) < 0);
+      if (pid > 0) {
+        usleep(2500); /* We need to die AFTER pidfile is written to... */
```

Comment:
Uh oh, magical constants. Isn't there a better way? What if it takes more than 2.5 milliseconds? Why don't we write the PID from the parent (fork returns the pid of the child to the parent on success).